app/main.py

```diff
@@ -68,7 +71,7 @@ def _parse_sp_file(self):
         with open(file_path, 'r') as sp_file:
             sp_data = json.load(sp_file)
             # retrieve the relevant values used to authenticate with Key Vault
-            self.tenant_id = sp_data['tenantId']
+            self.tenant_id = self._get_tenant_id()
```

Comment:
please make this dynamic search **optional**, and **disabled by default**, as this is an edge-case, and we should keep supporting using the explicit _tenant_id_ in the config file

Reply:
done. introduced an environment variable named `AUTO_DETECT_AAD_TENANT` and a value of `true` in that will enable dynamic detection.